Bottom: 3877d0fb505bce9c681935666b5a4872dbe825dd
Top:    9eeb7ee710176124e7df1828196a7d4c0710d01a
Author: Jon Smirl <jonsmirl@gmail.com>
Date:   2012-04-23 07:25:22 -0400

clean out val315x code


---

diff --git a/arch/arm/mach-lpc31xx/Kconfig b/arch/arm/mach-lpc31xx/Kconfig
index 7b0478d..04fb90a 100644
--- a/arch/arm/mach-lpc31xx/Kconfig
+++ b/arch/arm/mach-lpc31xx/Kconfig
@@ -16,17 +16,6 @@ config MACH_EA3152
 	help
 	  Say Y here if you are using Embedded Artist's EA3152 board
 	  
-config MACH_VAL3153
-	bool "val3153 board"
-	help
-	  Say Y here if you are using the NXP val3153 board
-
-config MACH_VAL3154
-	bool "val3154 board"
-	select LPC3152_AD
-	help
-	  Say Y here if you are using the NXP val3154 board
-	  
 endchoice
 
 config LPC3152_AD
diff --git a/arch/arm/mach-lpc31xx/Makefile b/arch/arm/mach-lpc31xx/Makefile
index 12522c0..d307a1a 100644
--- a/arch/arm/mach-lpc31xx/Makefile
+++ b/arch/arm/mach-lpc31xx/Makefile
@@ -9,10 +9,8 @@ obj-$(CONFIG_OF) += dt.o
 
 
 # Specific board support
-obj-$(CONFIG_MACH_VAL3153) += board-val3153.o leds.o
 obj-$(CONFIG_MACH_EA313X) += board-ea313x.o
 obj-$(CONFIG_MACH_EA3152) += board-ea313x.o
-obj-$(CONFIG_MACH_VAL3154) += board-val3154.o leds.o
 obj-$(CONFIG_LPC3152_AD) += psu.o
 
 # Power Management
diff --git a/arch/arm/mach-lpc31xx/dma.c b/arch/arm/mach-lpc31xx/dma.c
index e4bdcc1..389418d 100644
--- a/arch/arm/mach-lpc31xx/dma.c
+++ b/arch/arm/mach-lpc31xx/dma.c
@@ -324,7 +324,7 @@ int dma_read_counter (unsigned int chn, unsigned int * pcnt)
 	return 0;
 }
 
-int dma_write_counter (unsigned int chn, u32 cnt)
+int dma_write_counter (unsigned int chn, uint32_t cnt)
 {
 	if (chn >= DMA_MAX_CHANNELS || !dma_channels[chn].name) {
 		return -EINVAL;
@@ -458,9 +458,9 @@ int dma_request_specific_sg_channel (int chn, char *name, dma_cb_t cb,
 	return sg_higher_channel[chn];
 }
 
-int dma_prog_sg_channel(int chn, u32 dma_sg_list)
+int dma_prog_sg_channel(int chn, uint32_t dma_sg_list)
 {
-	u32 dma_config;
+	uint32_t dma_config;
 
 	if (chn >= DMA_MAX_CHANNELS)
 		return -EINVAL;
@@ -553,9 +553,9 @@ int dma_release_sg_channel (unsigned int chn){return 0;}
 int dma_prog_channel (unsigned int chn, dma_setup_t *dma_setup){return 0;}
 int dma_start_channel (unsigned int chn){return 0;}
 int dma_stop_channel (unsigned int chn){return 0;}
-int dma_prog_sg_channel(int chn, u32 dma_sg_list){return 0;}
+int dma_prog_sg_channel(int chn, uint32_t dma_sg_list){return 0;}
 int dma_set_irq_mask(unsigned int chn, int half_int, int fin_int){return 0;}
-int dma_write_counter (unsigned int chn, u32 cnt){return 0;}
+int dma_write_counter (unsigned int chn, uint32_t cnt){return 0;}
 int dma_stop_channel_sg (unsigned int chn){return 0;}
 int dma_channel_enabled(unsigned int chn){return 0;}
 int dma_current_state (unsigned int   chn, unsigned int * psrc, unsigned int * pdst, unsigned int * plen, unsigned int * pcfg, unsigned int * pena, unsigned int * pcnt){return 0;}
diff --git a/arch/arm/mach-lpc31xx/include/mach/registers.h b/arch/arm/mach-lpc31xx/include/mach/registers.h
index 4f4757e2..33b3f1d 100644
--- a/arch/arm/mach-lpc31xx/include/mach/registers.h
+++ b/arch/arm/mach-lpc31xx/include/mach/registers.h
@@ -69,95 +69,6 @@
 #define UART_ICR_REG      __REG (UART_PHYS + 0x24)
 #define UART_FDR_REG      __REG (UART_PHYS + 0x28)
 
-#if 0
-/***********************************************************************
- * SPI register definitions
- **********************************************************************/
-#define SPI_CONFIG_REG    __REG (SPI_PHYS + 0x00)
-#define SPI_SLV_ENAB_REG  __REG (SPI_PHYS + 0x04)
-#define SPI_TXF_FLUSH_REG __REG (SPI_PHYS + 0x08)
-#define SPI_FIFO_DATA_REG __REG (SPI_PHYS + 0x0C)
-#define SPI_NHP_POP_REG   __REG (SPI_PHYS + 0x10)
-#define SPI_NHP_MODE_REG  __REG (SPI_PHYS + 0x14)
-#define SPI_DMA_SET_REG   __REG (SPI_PHYS + 0x18)
-#define SPI_STS_REG       __REG (SPI_PHYS + 0x1C)
-#define SPI_HWINFO_REG    __REG (SPI_PHYS + 0x20)
-#define SPI_SLV_SET1_REG(slv) __REG (SPI_PHYS + 0x24 + (8 * slv))
-#define SPI_SLV_SET2_REG(slv) __REG (SPI_PHYS + 0x28 + (8 * slv))
-#define SPI_INT_TRSH_REG  __REG (SPI_PHYS + 0xFD4)
-#define SPI_INT_CLRE_REG  __REG (SPI_PHYS + 0xFD8)
-#define SPI_INT_SETE_REG  __REG (SPI_PHYS + 0xFDC)
-#define SPI_INT_STS_REG   __REG (SPI_PHYS + 0xFE0)
-#define SPI_INT_ENAB_REG  __REG (SPI_PHYS + 0xFE4)
-#define SPI_INT_CLRS_REG  __REG (SPI_PHYS + 0xFE8)
-#define SPI_INT_SETS_REG  __REG (SPI_PHYS + 0xFEC)
-#define SPI_MOD_ID_REG    __REG (SPI_PHYS + 0xFFC)
-
-/* SPI device contants */
-#define SPI_FIFO_DEPTH  64 /* 64 words (16bit) deep */
-#define SPI_NUM_SLAVES  3  /* number of slaves supported */
-#define SPI_MAX_DIV2    254
-#define SPI_MAX_DIVIDER 65024 /* = 254 * (255 + 1) */
-#define SPI_MIN_DIVIDER 2
-
-/* SPI Configuration register definitions (SPI_CONFIG_REG) */
-#define SPI_CFG_INTER_DLY(n)      _SBF(16, ((n) & 0xFFFF))
-#define SPI_CFG_INTER_DLY_GET(n)  (((n) >> 16) & 0xFFFF)
-#define SPI_CFG_UPDATE_EN         _BIT(7)
-#define SPI_CFG_SW_RESET          _BIT(6)
-#define SPI_CFG_SLAVE_DISABLE     _BIT(4)
-#define SPI_CFG_MULTI_SLAVE       _BIT(3)
-#define SPI_CFG_LOOPBACK          _BIT(2)
-#define SPI_CFG_SLAVE_MODE        _BIT(1)
-#define SPI_CFG_ENABLE            _BIT(0)
-
-/* SPI slave_enable register definitions (SPI_SLV_ENAB_REG) */
-#define SPI_SLV_EN(n)             _SBF(((n) << 1), 0x1)
-#define SPI_SLV_SUSPEND(n)        _SBF(((n) << 1), 0x3)
-
-/* SPI tx_fifo_flush register definitions (SPI_TXF_FLUSH_REG) */
-#define SPI_TXFF_FLUSH            _BIT(1)
-
-/* SPI dma_settings register definitions (SPI_DMA_SET_REG) */
-#define SPI_DMA_TX_EN             _BIT(1)
-#define SPI_DMA_RX_EN             _BIT(0)
-
-/* SPI status register definitions (SPI_STS_REG) */
-#define SPI_ST_SMS_BUSY           _BIT(5)
-#define SPI_ST_BUSY               _BIT(4)
-#define SPI_ST_RX_FF              _BIT(3)
-#define SPI_ST_RX_EMPTY           _BIT(2)
-#define SPI_ST_TX_FF              _BIT(1)
-#define SPI_ST_TX_EMPTY           _BIT(0)
-
-/* SPI slv_setting registers definitions (SPI_SLV_SET1_REG) */
-#define SPI_SLV1_INTER_TX_DLY(n)  _SBF(24, ((n) & 0xFF))
-#define SPI_SLV1_NUM_WORDS(n)     _SBF(16, ((n) & 0xFF))
-#define SPI_SLV1_CLK_PS(n)        _SBF(8, ((n) & 0xFF))
-#define SPI_SLV1_CLK_PS_GET(n)    (((n) >> 8) & 0xFF)
-#define SPI_SLV1_CLK_DIV1(n)      ((n) & 0xFF)
-#define SPI_SLV1_CLK_DIV1_GET(n)  ((n) & 0xFF)
-
-/* SPI slv_setting registers definitions (SPI_SLV_SET2_REG) */
-#define SPI_SLV2_PPCS_DLY(n)      _SBF(9, ((n) & 0xFF))
-#define SPI_SLV2_CS_HIGH          _BIT(8)
-#define SPI_SLV2_SSI_MODE         _BIT(7)
-#define SPI_SLV2_SPO              _BIT(6)
-#define SPI_SLV2_SPH              _BIT(5)
-#define SPI_SLV2_WD_SZ(n)         ((n) & 0x1F)
-
-/* SPI int_threshold registers definitions (SPI_INT_TRSH_REG) */
-#define SPI_INT_TSHLD_TX(n)       _SBF(8, ((n) & 0xFF))
-#define SPI_INT_TSHLD_RX(n)       ((n) & 0xFF)
-
-/* SPI intterrupt registers definitions ( SPI_INT_xxx) */
-#define SPI_SMS_INT               _BIT(4)
-#define SPI_TX_INT                _BIT(3)
-#define SPI_RX_INT                _BIT(2)
-#define SPI_TO_INT                _BIT(1)
-#define SPI_OVR_INT               _BIT(0)
-#define SPI_ALL_INTS              (SPI_SMS_INT | SPI_TX_INT | SPI_RX_INT | SPI_TO_INT | SPI_OVR_INT)
-#endif
 
 /***********************************************************************
  * ADC_REG register definitions
@@ -197,24 +108,6 @@
 #define SYS_MUX_GPIO_MCI    __REG (SYS_PHYS + 0x94)
 #define SYS_MUX_NAND_MCI    __REG (SYS_PHYS + 0x98)
 
-/***********************************************************************
- * GPIO register definitions
- **********************************************************************/
-#if 0
-#define GPIO_STATE(port)     __REG (GPIO_PHYS + (port) + 0x00)
-#define GPIO_STATE_M0(port)  __REG (GPIO_PHYS + (port) + 0x10)
-#define GPIO_M0_SET(port)    __REG (GPIO_PHYS + (port) + 0x14)
-#define GPIO_M0_RESET(port)  __REG (GPIO_PHYS + (port) + 0x18)
-#define GPIO_STATE_M1(port)  __REG (GPIO_PHYS + (port) + 0x20)
-#define GPIO_M1_SET(port)    __REG (GPIO_PHYS + (port) + 0x24)
-#define GPIO_M1_RESET(port)  __REG (GPIO_PHYS + (port) + 0x28)
-#endif
-
-#define GPIO_OUT_LOW(port, pin)  do { GPIO_M1_SET(port) = pin; GPIO_M0_RESET(port) = pin;} while(0)
-#define GPIO_OUT_HIGH(port, pin) do { GPIO_M1_SET(port) = pin; GPIO_M0_SET(port) = pin;} while(0)
-#define GPIO_IN(port, pin)       do { GPIO_M1_RESET(port) = pin; GPIO_M0_RESET(port) = pin;} while(0)
-#define GPIO_DRV_IP(port, pin)   do { GPIO_M1_RESET(port) = pin; GPIO_M0_SET(port) = pin;} while(0)
-
 #define IOCONF_EBI_MCI       (0x000)
 #define IOCONF_EBI_I2STX_0   (0x040)
 #define IOCONF_CGU           (0x080)
diff --git a/arch/arm/mach-lpc31xx/leds.c b/arch/arm/mach-lpc31xx/leds.c
deleted file mode 100644
index 26ab4ca..0000000
--- a/arch/arm/mach-lpc31xx/leds.c
+++ /dev/null
@@ -1,102 +0,0 @@
-/*  linux/arch/arm/mach-lpc31xx/leds.c
- *
- *  Author:	Durgesh Pattamatta
- *  Copyright (C) 2009 NXP semiconductors
- *
- * LED driver for val3153-based boards.
- *
- * This program is free software; you can redistribute it and/or modify
- * it under the terms of the GNU General Public License as published by
- * the Free Software Foundation; either version 2 of the License, or
- * (at your option) any later version.
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License
- * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
- *
- */
-
-#include <linux/kernel.h>
-#include <linux/module.h>
-#include <linux/init.h>
-
-#include <asm/leds.h>
-#include <mach/gpio.h>
-
-
-static inline void val3153_led_on(unsigned int led)
-{
-	gpio_set_value(led, 0);
-}
-
-static inline void val3153_led_off(unsigned int led)
-{
-	gpio_set_value(led, 1);
-}
-
-static inline void val3153_led_toggle(unsigned int led)
-{
-	unsigned long is_off = gpio_get_value(led);
-	if (is_off)
-		val3153_led_on(led);
-	else
-		val3153_led_off(led);
-}
-
-
-/*
- * Handle LED events.
- */
-static void val3153_leds_event(led_event_t evt)
-{
-	unsigned long flags;
-
-	local_irq_save(flags);
-
-	switch(evt) {
-	case led_start:		/* System startup */
-		val3153_led_on(GPIO_GPIO0);
-		break;
-
-	case led_stop:		/* System stop / suspend */
-		val3153_led_off(GPIO_GPIO0);
-		break;
-
-#ifdef CONFIG_LEDS_TIMER
-	case led_timer:		/* Every 50 timer ticks */
-		val3153_led_toggle(GPIO_GPIO2);
-		break;
-#endif
-
-#ifdef CONFIG_LEDS_CPU
-	case led_idle_start:	/* Entering idle state */
-		val3153_led_off(GPIO_GPIO0);
-		break;
-
-	case led_idle_end:	/* Exit idle state */
-		val3153_led_on(GPIO_GPIO0);
-		break;
-#endif
-
-	default:
-		break;
-	}
-
-	local_irq_restore(flags);
-}
-
-
-int __init leds_init(void)
-{
-	leds_event = val3153_leds_event;
-
-	leds_event(led_start);
-	return 0;
-}
-
-__initcall(leds_init);
diff --git a/arch/arm/mach-lpc31xx/psu.c b/arch/arm/mach-lpc31xx/psu.c
index 5385fea..b8a8bff 100644
--- a/arch/arm/mach-lpc31xx/psu.c
+++ b/arch/arm/mach-lpc31xx/psu.c
@@ -1,4 +1,4 @@
-/*  linux/arch/arm/mach-lpc31xx/leds.c
+/*  linux/arch/arm/mach-lpc31xx/psu.c
  *
  *  Author:	Durgesh Pattamatta
  *  Copyright (C) 2009 NXP semiconductors
@@ -63,9 +63,9 @@ static struct psu_data g_pca_data;
  */
 struct i2c_client *lpc315x_ad_get_i2c_client_struct(void)
 {
-	/* Check if psu_data structure is initialised */
+	/* Check if psu_data structure is initialized */
 	if(!g_pca_data.client) {
-		printk(KERN_ERR "I2C not initialised \r\n");
+		printk(KERN_ERR "I2C not initialized \r\n");
 		return NULL;
 	}
 
@@ -80,7 +80,7 @@ static ssize_t psu_show(struct device *dev, struct device_attribute *attr,
 	struct sensor_device_attribute *psa = to_sensor_dev_attr(attr);
 	struct i2c_client *client = to_i2c_client(dev);
 	char reg_adr[2];
-	u32 reg_val = 0;
+	uint32_t reg_val = 0;
 
 	reg_adr[0] = (psa->index >> 8) & 0xFF;
 	reg_adr[1] = psa->index & 0xFF;
diff --git a/arch/arm/mach-lpc31xx/time.c b/arch/arm/mach-lpc31xx/time.c
index d17779d..b36894c 100644
--- a/arch/arm/mach-lpc31xx/time.c
+++ b/arch/arm/mach-lpc31xx/time.c
@@ -78,7 +78,7 @@ static struct irqaction lpc31xx_timer_irq = {
  */
 static unsigned long lpc31xx_gettimeoffset(void)
 {
-	u32 elapsed = LATCH - timer_read(TIMER_VALUE);
+	uint32_t elapsed = LATCH - timer_read(TIMER_VALUE);
 	return ((elapsed * 100) / (XTAL_CLOCK / 20000));
 }
 
diff --git a/arch/arm/mach-lpc31xx/usb.c b/arch/arm/mach-lpc31xx/usb.c
index 4596c65..45ad7e6 100644
--- a/arch/arm/mach-lpc31xx/usb.c
+++ b/arch/arm/mach-lpc31xx/usb.c
@@ -102,8 +102,8 @@
 #if defined(CONFIG_USB_GADGET_FSL_USB2) || defined(CONFIG_USB_OTG)
 static struct resource lpc31xx_usb_resource[] = {
 	[0] = {
-		.start = (u32) (USBOTG_PHYS),
-		.end   = (u32) (USBOTG_PHYS + SZ_4K),
+		.start = (uint32_t) (USBOTG_PHYS),
+		.end   = (uint32_t) (USBOTG_PHYS + SZ_4K),
 		.flags = IORESOURCE_MEM,
 	},
 	[1] = {
@@ -188,8 +188,8 @@ static void lpc31xx_vbusen_timer(unsigned long data)
 int __init usbotg_init(void)
 {
 	int over;
-	u32 bank = EVT_GET_BANK(EVT_usb_atx_pll_lock);
-	u32 bit_pos = EVT_usb_atx_pll_lock & 0x1F;
+	uint32_t bank = EVT_GET_BANK(EVT_usb_atx_pll_lock);
+	uint32_t bit_pos = EVT_usb_atx_pll_lock & 0x1F;
 	int retval = 0;
 
 	/* enable USB to AHB clock */
diff --git a/arch/arm/tools/mach-types b/arch/arm/tools/mach-types
index 6e4e0a5..e60254e 100644
--- a/arch/arm/tools/mach-types
+++ b/arch/arm/tools/mach-types
@@ -1206,8 +1206,6 @@ elite_ulk		MACH_ELITE_ULK		ELITE_ULK		3888
 pov2			MACH_POV2		POV2			3889
 ipod_touch_2g		MACH_IPOD_TOUCH_2G	IPOD_TOUCH_2G		3890
 da850_pqab		MACH_DA850_PQAB		DA850_PQAB		3891
-val3153			MACH_VAL3153		VAL3153			9999
-val3154			MACH_VAL3154		VAL3154			9997
 ea3152			MACH_EA3152		EA3152			9996
 
 
diff --git a/drivers/net/ethernet/cirrus/cs89x0.c b/drivers/net/ethernet/cirrus/cs89x0.c
index f7e0595..b9406cb 100644
--- a/drivers/net/ethernet/cirrus/cs89x0.c
+++ b/drivers/net/ethernet/cirrus/cs89x0.c
@@ -321,12 +321,6 @@ struct net_device * __init cs89x0_probe(int unit)
 	if (net_debug)
 		printk("cs89x0:cs89x0_probe(0x%x)\n", io);
 
-#ifdef CONFIG_MACH_VAL3153
-	if(unit > 0) {
-		err = -ENODEV;
-		goto out;
-	}
-#endif
 	if (io > 0x1ff)	{	/* Check a single specified location. */
 		err = cs89x0_probe1(dev, io, 0);
 	} else if (io != 0) {	/* Don't probe at all. */
@@ -375,37 +369,6 @@ writeword(unsigned long base_addr, int portno, u16 value)
 {
 	__raw_writel(value, base_addr + (portno << 1));
 }
-#elif defined(CONFIG_ARCH_PNX010X)
-static u16
-readword(unsigned long base_addr, int portno)
-{
-	return inw(base_addr + (portno << 1));
-}
-
-static void
-writeword(unsigned long base_addr, int portno, u16 value)
-{
-	outw(value, base_addr + (portno << 1));
-}
-#elif defined(CONFIG_MACH_VAL3153)
-static u16
-readword(unsigned long base_addr, int portno)
-{
-	u16 v;
-	CS8900_IOBARRIER;
-	v = inw(base_addr + portno);
-	CS8900_IOBARRIER;
-	return v;
-}
-
-static void
-writeword(unsigned long base_addr, int portno, u16 value)
-{
-	CS8900_IOBARRIER;
-	outw(value, base_addr + portno);
-	CS8900_IOBARRIER;
-}
-
 #else
 static u16
 readword(unsigned long base_addr, int portno)
@@ -578,12 +541,6 @@ cs89x0_probe1(struct net_device *dev, unsigned long ioaddr, int modular)
 		goto out1;
 	}
 
-#if defined(CONFIG_MACH_VAL3153)
-	/* truely reset the chip */
-	writeword(ioaddr, ADD_PORT, 0x0114);
-	writeword(ioaddr, DATA_PORT, 0x0040);
-#endif
-
 	/* if they give us an odd I/O address, then do ONE write to
            the address port, to get it back to address zero, where we
            expect to find the EISA signature word. An IO with a base of 0x3
@@ -649,12 +606,9 @@ cs89x0_probe1(struct net_device *dev, unsigned long ioaddr, int modular)
 	   the driver will always do *something* instead of complain that
 	   adapter_cnf is 0. */
 
-/* quick hack for VAL3153 boards to reuse the mac address set by boot loader */
-#if !defined(CONFIG_MACH_VAL3153)
+
         if ((readreg(dev, PP_SelfST) & (EEPROM_OK | EEPROM_PRESENT)) ==
-	      (EEPROM_OK|EEPROM_PRESENT)) 
-#endif
-	{
+	      (EEPROM_OK|EEPROM_PRESENT)) {
 	        /* Load the MAC. */
 		for (i=0; i < ETH_ALEN/2; i++) {
 	                unsigned int Addr;
diff --git a/dt_config b/dt_config
index 9a449e2..0998c92 100644
--- a/dt_config
+++ b/dt_config
@@ -1,6 +1,6 @@
 #
 # Automatically generated file; DO NOT EDIT.
-# Linux/arm 3.3.0 Kernel Configuration
+# Linux/arm 3.4.0-rc4 Kernel Configuration
 #
 CONFIG_ARM=y
 CONFIG_SYS_SUPPORTS_APM_EMULATION=y
@@ -36,9 +36,11 @@ CONFIG_LOCALVERSION=""
 # CONFIG_LOCALVERSION_AUTO is not set
 CONFIG_HAVE_KERNEL_GZIP=y
 CONFIG_HAVE_KERNEL_LZMA=y
+CONFIG_HAVE_KERNEL_XZ=y
 CONFIG_HAVE_KERNEL_LZO=y
 CONFIG_KERNEL_GZIP=y
 # CONFIG_KERNEL_LZMA is not set
+# CONFIG_KERNEL_XZ is not set
 # CONFIG_KERNEL_LZO is not set
 CONFIG_DEFAULT_HOSTNAME="(none)"
 # CONFIG_SWAP is not set
@@ -55,17 +57,15 @@ CONFIG_HAVE_GENERIC_HARDIRQS=y
 # IRQ subsystem
 #
 CONFIG_GENERIC_HARDIRQS=y
-CONFIG_HAVE_SPARSE_IRQ=y
 CONFIG_GENERIC_IRQ_SHOW=y
 CONFIG_IRQ_DOMAIN=y
-# CONFIG_SPARSE_IRQ is not set
+CONFIG_IRQ_DOMAIN_DEBUG=y
 
 #
 # RCU Subsystem
 #
 CONFIG_TINY_RCU=y
 # CONFIG_PREEMPT_RCU is not set
-# CONFIG_RCU_TRACE is not set
 # CONFIG_TREE_RCU_TRACE is not set
 # CONFIG_IKCONFIG is not set
 CONFIG_LOG_BUF_SHIFT=14
@@ -124,11 +124,13 @@ CONFIG_SLAB=y
 # CONFIG_PROFILING is not set
 CONFIG_HAVE_OPROFILE=y
 # CONFIG_KPROBES is not set
+# CONFIG_JUMP_LABEL is not set
 CONFIG_HAVE_KPROBES=y
 CONFIG_HAVE_KRETPROBES=y
 CONFIG_HAVE_REGS_AND_STACK_ACCESS_API=y
 CONFIG_HAVE_CLK=y
 CONFIG_HAVE_DMA_API_DEBUG=y
+CONFIG_HAVE_ARCH_JUMP_LABEL=y
 
 #
 # GCOV-based kernel profiling
@@ -170,7 +172,6 @@ CONFIG_DEFAULT_IOSCHED="noop"
 # CONFIG_INLINE_SPIN_LOCK_BH is not set
 # CONFIG_INLINE_SPIN_LOCK_IRQ is not set
 # CONFIG_INLINE_SPIN_LOCK_IRQSAVE is not set
-CONFIG_INLINE_SPIN_UNLOCK=y
 # CONFIG_INLINE_SPIN_UNLOCK_BH is not set
 CONFIG_INLINE_SPIN_UNLOCK_IRQ=y
 # CONFIG_INLINE_SPIN_UNLOCK_IRQRESTORE is not set
@@ -239,7 +240,7 @@ CONFIG_MMU=y
 # CONFIG_ARCH_SHMOBILE is not set
 # CONFIG_ARCH_RPC is not set
 # CONFIG_ARCH_SA1100 is not set
-# CONFIG_ARCH_S3C2410 is not set
+# CONFIG_ARCH_S3C24XX is not set
 # CONFIG_ARCH_S3C64XX is not set
 # CONFIG_ARCH_S5P64X0 is not set
 # CONFIG_ARCH_S5PC100 is not set
@@ -262,12 +263,10 @@ CONFIG_ARCH_LPC31XX=y
 #
 
 #
-# LPC313x Implementations
+# LPC31xx Implementations
 #
 CONFIG_MACH_EA313X=y
 # CONFIG_MACH_EA3152 is not set
-# CONFIG_MACH_VAL3153 is not set
-# CONFIG_MACH_VAL3154 is not set
 # CONFIG_HZ_100 is not set
 CONFIG_HZ_250=y
 # CONFIG_HZ_300 is not set
@@ -495,6 +494,8 @@ CONFIG_IEEE802154=y
 # CONFIG_BATMAN_ADV is not set
 # CONFIG_OPENVSWITCH is not set
 CONFIG_BQL=y
+CONFIG_HAVE_BPF_JIT=y
+# CONFIG_BPF_JIT is not set
 
 #
 # Network testing
@@ -526,7 +527,6 @@ CONFIG_MAC80211_RC_MINSTREL_HT=y
 CONFIG_MAC80211_RC_DEFAULT_MINSTREL=y
 CONFIG_MAC80211_RC_DEFAULT="minstrel_ht"
 # CONFIG_MAC80211_MESH is not set
-# CONFIG_MAC80211_LEDS is not set
 # CONFIG_MAC80211_DEBUGFS is not set
 # CONFIG_MAC80211_DEBUG_MENU is not set
 # CONFIG_WIMAX is not set
@@ -556,98 +556,9 @@ CONFIG_EXTRA_FIRMWARE=""
 # CONFIG_GENERIC_CPU_DEVICES is not set
 CONFIG_REGMAP=y
 CONFIG_REGMAP_I2C=y
-CONFIG_REGMAP_SPI=y
 # CONFIG_DMA_SHARED_BUFFER is not set
 # CONFIG_CONNECTOR is not set
-CONFIG_MTD=y
-# CONFIG_MTD_TESTS is not set
-# CONFIG_MTD_REDBOOT_PARTS is not set
-# CONFIG_MTD_CMDLINE_PARTS is not set
-# CONFIG_MTD_AFS_PARTS is not set
-# CONFIG_MTD_OF_PARTS is not set
-# CONFIG_MTD_AR7_PARTS is not set
-
-#
-# User Modules And Translation Layers
-#
-CONFIG_MTD_CHAR=y
-CONFIG_MTD_BLKDEVS=y
-CONFIG_MTD_BLOCK=y
-# CONFIG_FTL is not set
-# CONFIG_NFTL is not set
-# CONFIG_INFTL is not set
-# CONFIG_RFD_FTL is not set
-# CONFIG_SSFDC is not set
-# CONFIG_SM_FTL is not set
-# CONFIG_MTD_OOPS is not set
-
-#
-# RAM/ROM/Flash chip drivers
-#
-# CONFIG_MTD_CFI is not set
-# CONFIG_MTD_JEDECPROBE is not set
-CONFIG_MTD_MAP_BANK_WIDTH_1=y
-CONFIG_MTD_MAP_BANK_WIDTH_2=y
-CONFIG_MTD_MAP_BANK_WIDTH_4=y
-# CONFIG_MTD_MAP_BANK_WIDTH_8 is not set
-# CONFIG_MTD_MAP_BANK_WIDTH_16 is not set
-# CONFIG_MTD_MAP_BANK_WIDTH_32 is not set
-CONFIG_MTD_CFI_I1=y
-CONFIG_MTD_CFI_I2=y
-# CONFIG_MTD_CFI_I4 is not set
-# CONFIG_MTD_CFI_I8 is not set
-# CONFIG_MTD_RAM is not set
-# CONFIG_MTD_ROM is not set
-# CONFIG_MTD_ABSENT is not set
-
-#
-# Mapping drivers for chip access
-#
-# CONFIG_MTD_COMPLEX_MAPPINGS is not set
-# CONFIG_MTD_PLATRAM is not set
-
-#
-# Self-contained MTD device drivers
-#
-CONFIG_MTD_DATAFLASH=y
-# CONFIG_MTD_DATAFLASH_WRITE_VERIFY is not set
-# CONFIG_MTD_DATAFLASH_OTP is not set
-CONFIG_MTD_M25P80=y
-CONFIG_M25PXX_USE_FAST_READ=y
-# CONFIG_MTD_SST25L is not set
-# CONFIG_MTD_SLRAM is not set
-# CONFIG_MTD_PHRAM is not set
-# CONFIG_MTD_MTDRAM is not set
-# CONFIG_MTD_BLOCK2MTD is not set
-
-#
-# Disk-On-Chip Device Drivers
-#
-# CONFIG_MTD_DOC2000 is not set
-# CONFIG_MTD_DOC2001 is not set
-# CONFIG_MTD_DOC2001PLUS is not set
-# CONFIG_MTD_DOCG3 is not set
-CONFIG_MTD_NAND_ECC=y
-# CONFIG_MTD_NAND_ECC_SMC is not set
-CONFIG_MTD_NAND=y
-# CONFIG_MTD_NAND_VERIFY_WRITE is not set
-# CONFIG_MTD_NAND_ECC_BCH is not set
-# CONFIG_MTD_SM_COMMON is not set
-# CONFIG_MTD_NAND_MUSEUM_IDS is not set
-# CONFIG_MTD_NAND_GPIO is not set
-CONFIG_MTD_NAND_IDS=y
-# CONFIG_MTD_NAND_DISKONCHIP is not set
-# CONFIG_MTD_NAND_NANDSIM is not set
-# CONFIG_MTD_NAND_PLATFORM is not set
-# CONFIG_MTD_ALAUDA is not set
-CONFIG_MTD_NAND_LPC313X=y
-# CONFIG_MTD_ONENAND is not set
-
-#
-# LPDDR flash memory drivers
-#
-# CONFIG_MTD_LPDDR is not set
-# CONFIG_MTD_UBI is not set
+# CONFIG_MTD is not set
 CONFIG_DTC=y
 CONFIG_OF=y
 
@@ -664,7 +575,6 @@ CONFIG_OF_DEVICE=y
 CONFIG_OF_GPIO=y
 CONFIG_OF_I2C=y
 CONFIG_OF_NET=y
-CONFIG_OF_SPI=y
 # CONFIG_PARPORT is not set
 CONFIG_BLK_DEV=y
 # CONFIG_BLK_DEV_COW_COMMON is not set
@@ -676,7 +586,6 @@ CONFIG_BLK_DEV_LOOP_MIN_COUNT=8
 # DRBD disabled because PROC_FS, INET or CONNECTOR not selected
 #
 # CONFIG_BLK_DEV_NBD is not set
-# CONFIG_BLK_DEV_UB is not set
 CONFIG_BLK_DEV_RAM=y
 CONFIG_BLK_DEV_RAM_COUNT=16
 CONFIG_BLK_DEV_RAM_SIZE=8192
@@ -703,7 +612,6 @@ CONFIG_BLK_DEV_XIP=y
 # CONFIG_SENSORS_APDS990X is not set
 # CONFIG_HMC6352 is not set
 # CONFIG_DS1682 is not set
-# CONFIG_TI_DAC7512 is not set
 # CONFIG_BMP085 is not set
 # CONFIG_USB_SWITCH_FSA9480 is not set
 # CONFIG_C2PORT is not set
@@ -712,27 +620,24 @@ CONFIG_BLK_DEV_XIP=y
 # EEPROM support
 #
 # CONFIG_EEPROM_AT24 is not set
-# CONFIG_EEPROM_AT25 is not set
 # CONFIG_EEPROM_LEGACY is not set
 # CONFIG_EEPROM_MAX6875 is not set
 # CONFIG_EEPROM_93CX6 is not set
-# CONFIG_EEPROM_93XX46 is not set
 # CONFIG_IWMC3200TOP is not set
 
 #
 # Texas Instruments shared transport line discipline
 #
 # CONFIG_TI_ST is not set
-# CONFIG_SENSORS_LIS3_SPI is not set
 # CONFIG_SENSORS_LIS3_I2C is not set
 
 #
 # Altera FPGA firmware download module
 #
 # CONFIG_ALTERA_STAPL is not set
-CONFIG_NEW_LPC313x_IO=y
-CONFIG_LPC313x_ADC=y
-CONFIG_LPC313x_PWM=y
+CONFIG_NEW_LPC31xx_IO=y
+CONFIG_LPC31xx_ADC=y
+CONFIG_LPC31xx_PWM=y
 
 #
 # SCSI device support
@@ -804,6 +709,7 @@ CONFIG_NET_VENDOR_BROADCOM=y
 # CONFIG_B44 is not set
 # CONFIG_NET_CALXEDA_XGMAC is not set
 CONFIG_NET_VENDOR_CHELSIO=y
+# CONFIG_NET_VENDOR_CIRRUS is not set
 CONFIG_DM9000=y
 # CONFIG_DM9000_FORCE_SIMPLE_PHY_POLL is not set
 # CONFIG_DNET is not set
@@ -814,10 +720,7 @@ CONFIG_NET_VENDOR_INTEL=y
 CONFIG_NET_VENDOR_I825XX=y
 CONFIG_NET_VENDOR_MARVELL=y
 CONFIG_NET_VENDOR_MICREL=y
-# CONFIG_KS8851 is not set
 # CONFIG_KS8851_MLL is not set
-CONFIG_NET_VENDOR_MICROCHIP=y
-# CONFIG_ENC28J60 is not set
 CONFIG_NET_VENDOR_NATSEMI=y
 CONFIG_NET_VENDOR_8390=y
 # CONFIG_AX88796 is not set
@@ -831,25 +734,10 @@ CONFIG_NET_VENDOR_SMSC=y
 CONFIG_NET_VENDOR_STMICRO=y
 # CONFIG_STMMAC_ETH is not set
 # CONFIG_PHYLIB is not set
-# CONFIG_MICREL_KS8995MA is not set
 # CONFIG_PPP is not set
 # CONFIG_SLIP is not set
-
-#
-# USB Network Adapters
-#
-# CONFIG_USB_CATC is not set
-# CONFIG_USB_KAWETH is not set
-# CONFIG_USB_PEGASUS is not set
-# CONFIG_USB_RTL8150 is not set
-# CONFIG_USB_USBNET is not set
-# CONFIG_USB_IPHETH is not set
 CONFIG_WLAN=y
 # CONFIG_LIBERTAS_THINFIRM is not set
-# CONFIG_AT76C50X_USB is not set
-# CONFIG_USB_ZD1201 is not set
-# CONFIG_USB_NET_RNDIS_WLAN is not set
-# CONFIG_RTL8187 is not set
 # CONFIG_MAC80211_HWSIM is not set
 # CONFIG_ATH_COMMON is not set
 # CONFIG_B43 is not set
@@ -860,24 +748,8 @@ CONFIG_WLAN=y
 # CONFIG_LIBERTAS is not set
 # CONFIG_P54_COMMON is not set
 CONFIG_RT2X00=y
-CONFIG_RT2500USB=y
-CONFIG_RT73USB=y
-CONFIG_RT2800USB=y
-CONFIG_RT2800USB_RT33XX=y
-CONFIG_RT2800USB_RT35XX=y
-# CONFIG_RT2800USB_RT53XX is not set
-# CONFIG_RT2800USB_UNKNOWN is not set
-CONFIG_RT2800_LIB=y
-CONFIG_RT2X00_LIB_USB=y
-CONFIG_RT2X00_LIB=y
-CONFIG_RT2X00_LIB_FIRMWARE=y
-CONFIG_RT2X00_LIB_CRYPTO=y
-CONFIG_RT2X00_LIB_LEDS=y
-# CONFIG_RT2X00_DEBUG is not set
-# CONFIG_RTL8192CU is not set
 # CONFIG_WL1251 is not set
 # CONFIG_WL12XX_MENU is not set
-# CONFIG_ZD1211RW is not set
 # CONFIG_MWIFIEX is not set
 
 #
@@ -885,7 +757,6 @@ CONFIG_RT2X00_LIB_LEDS=y
 #
 # CONFIG_WAN is not set
 # CONFIG_ISDN is not set
-# CONFIG_PHONE is not set
 
 #
 # Input device support
@@ -914,18 +785,18 @@ CONFIG_INPUT_MOUSEDEV_SCREEN_Y=768
 # CONFIG_INPUT_JOYSTICK is not set
 # CONFIG_INPUT_TABLET is not set
 CONFIG_INPUT_TOUCHSCREEN=y
-CONFIG_TOUCHSCREEN_ADS7846=y
-# CONFIG_TOUCHSCREEN_AD7877 is not set
 # CONFIG_TOUCHSCREEN_AD7879 is not set
 # CONFIG_TOUCHSCREEN_ATMEL_MXT is not set
 # CONFIG_TOUCHSCREEN_AUO_PIXCIR is not set
 # CONFIG_TOUCHSCREEN_BU21013 is not set
 # CONFIG_TOUCHSCREEN_CY8CTMG110 is not set
+# CONFIG_TOUCHSCREEN_CYTTSP_CORE is not set
 # CONFIG_TOUCHSCREEN_DYNAPRO is not set
 # CONFIG_TOUCHSCREEN_HAMPSHIRE is not set
 # CONFIG_TOUCHSCREEN_EETI is not set
 # CONFIG_TOUCHSCREEN_EGALAX is not set
 # CONFIG_TOUCHSCREEN_FUJITSU is not set
+# CONFIG_TOUCHSCREEN_ILI210X is not set
 # CONFIG_TOUCHSCREEN_GUNZE is not set
 # CONFIG_TOUCHSCREEN_ELO is not set
 # CONFIG_TOUCHSCREEN_WACOM_W8001 is not set
@@ -938,10 +809,8 @@ CONFIG_TOUCHSCREEN_ADS7846=y
 # CONFIG_TOUCHSCREEN_TOUCHRIGHT is not set
 # CONFIG_TOUCHSCREEN_TOUCHWIN is not set
 # CONFIG_TOUCHSCREEN_PIXCIR is not set
-# CONFIG_TOUCHSCREEN_USB_COMPOSITE is not set
 # CONFIG_TOUCHSCREEN_TOUCHIT213 is not set
 # CONFIG_TOUCHSCREEN_TSC_SERIO is not set
-# CONFIG_TOUCHSCREEN_TSC2005 is not set
 # CONFIG_TOUCHSCREEN_TSC2007 is not set
 # CONFIG_TOUCHSCREEN_W90X900 is not set
 # CONFIG_TOUCHSCREEN_ST1232 is not set
@@ -980,15 +849,12 @@ CONFIG_SERIAL_8250_RUNTIME_UARTS=1
 #
 # Non-8250 serial port support
 #
-# CONFIG_SERIAL_MAX3100 is not set
-# CONFIG_SERIAL_MAX3107 is not set
 CONFIG_SERIAL_CORE=y
 CONFIG_SERIAL_CORE_CONSOLE=y
 # CONFIG_SERIAL_OF_PLATFORM is not set
 # CONFIG_SERIAL_TIMBERDALE is not set
 # CONFIG_SERIAL_ALTERA_JTAGUART is not set
 # CONFIG_SERIAL_ALTERA_UART is not set
-# CONFIG_SERIAL_IFX6X60 is not set
 # CONFIG_SERIAL_XILINX_PS_UART is not set
 # CONFIG_TTY_PRINTK is not set
 # CONFIG_HVC_DCC is not set
@@ -1000,8 +866,8 @@ CONFIG_SERIAL_CORE_CONSOLE=y
 # CONFIG_RAMOOPS is not set
 CONFIG_I2C=y
 CONFIG_I2C_BOARDINFO=y
-# CONFIG_I2C_COMPAT is not set
-CONFIG_I2C_CHARDEV=y
+CONFIG_I2C_COMPAT=y
+# CONFIG_I2C_CHARDEV is not set
 # CONFIG_I2C_MUX is not set
 CONFIG_I2C_HELPER_AUTO=y
 
@@ -1024,10 +890,8 @@ CONFIG_I2C_PNX=y
 #
 # External I2C/SMBus adapter drivers
 #
-# CONFIG_I2C_DIOLAN_U2C is not set
 # CONFIG_I2C_PARPORT_LIGHT is not set
 # CONFIG_I2C_TAOS_EVM is not set
-# CONFIG_I2C_TINY_USB is not set
 
 #
 # Other I2C/SMBus bus drivers
@@ -1036,27 +900,8 @@ CONFIG_I2C_PNX=y
 # CONFIG_I2C_DEBUG_CORE is not set
 # CONFIG_I2C_DEBUG_ALGO is not set
 # CONFIG_I2C_DEBUG_BUS is not set
-CONFIG_SPI=y
-# CONFIG_SPI_DEBUG is not set
-CONFIG_SPI_MASTER=y
-
-#
-# SPI Master Controller Drivers
-#
-# CONFIG_SPI_ALTERA is not set
-# CONFIG_SPI_BITBANG is not set
-# CONFIG_SPI_GPIO is not set
-# CONFIG_SPI_OC_TINY is not set
-# CONFIG_SPI_PXA2XX_PCI is not set
-# CONFIG_SPI_XILINX is not set
-CONFIG_SPI_LPC313X=y
-# CONFIG_SPI_DESIGNWARE is not set
-
-#
-# SPI Protocol Masters
-#
-CONFIG_SPI_SPIDEV=y
-# CONFIG_SPI_TLE62X0 is not set
+# CONFIG_SPI is not set
+# CONFIG_HSI is not set
 
 #
 # PPS support
@@ -1100,10 +945,7 @@ CONFIG_GPIO_SYSFS=y
 #
 # SPI GPIO expanders:
 #
-# CONFIG_GPIO_MAX7301 is not set
 # CONFIG_GPIO_MCP23S08 is not set
-# CONFIG_GPIO_MC33880 is not set
-# CONFIG_GPIO_74X164 is not set
 
 #
 # AC97 GPIO expanders:
@@ -1127,11 +969,6 @@ CONFIG_WATCHDOG=y
 # CONFIG_DW_WATCHDOG is not set
 # CONFIG_MAX63XX_WATCHDOG is not set
 CONFIG_LPC31XX_WATCHDOG=y
-
-#
-# USB-based Watchdog Cards
-#
-# CONFIG_USBPCWATCHDOG is not set
 CONFIG_SSB_POSSIBLE=y
 
 #
@@ -1158,11 +995,12 @@ CONFIG_BCMA_POSSIBLE=y
 # CONFIG_TPS6105X is not set
 # CONFIG_TPS65010 is not set
 # CONFIG_TPS6507X is not set
+# CONFIG_MFD_TPS65217 is not set
 # CONFIG_MFD_TPS6586X is not set
 # CONFIG_MFD_TPS65910 is not set
 # CONFIG_MFD_TPS65912_I2C is not set
-# CONFIG_MFD_TPS65912_SPI is not set
 # CONFIG_TWL4030_CORE is not set
+# CONFIG_TWL6040_CORE is not set
 # CONFIG_MFD_STMPE is not set
 # CONFIG_MFD_TC3589X is not set
 # CONFIG_MFD_TMIO is not set
@@ -1170,7 +1008,6 @@ CONFIG_BCMA_POSSIBLE=y
 # CONFIG_MFD_TC6387XB is not set
 # CONFIG_MFD_TC6393XB is not set
 # CONFIG_PMIC_DA903X is not set
-# CONFIG_MFD_DA9052_SPI is not set
 # CONFIG_MFD_DA9052_I2C is not set
 # CONFIG_PMIC_ADP5520 is not set
 # CONFIG_MFD_MAX8925 is not set
@@ -1179,15 +1016,14 @@ CONFIG_BCMA_POSSIBLE=y
 # CONFIG_MFD_S5M_CORE is not set
 # CONFIG_MFD_WM8400 is not set
 # CONFIG_MFD_WM831X_I2C is not set
-# CONFIG_MFD_WM831X_SPI is not set
 # CONFIG_MFD_WM8350_I2C is not set
 # CONFIG_MFD_WM8994 is not set
 # CONFIG_MFD_PCF50633 is not set
-# CONFIG_MFD_MC13XXX is not set
 # CONFIG_ABX500_CORE is not set
-# CONFIG_EZX_PCAP is not set
 # CONFIG_MFD_WL1273_CORE is not set
+# CONFIG_MFD_TPS65090 is not set
 # CONFIG_MFD_AAT2870_CORE is not set
+# CONFIG_MFD_RC5T583 is not set
 # CONFIG_REGULATOR is not set
 # CONFIG_MEDIA_SUPPORT is not set
 
@@ -1222,12 +1058,11 @@ CONFIG_FB_DEFERRED_IO=y
 # Frame buffer hardware drivers
 #
 # CONFIG_FB_S1D13XXX is not set
-# CONFIG_FB_SMSCUFX is not set
-# CONFIG_FB_UDL is not set
 # CONFIG_FB_VIRTUAL is not set
 # CONFIG_FB_METRONOME is not set
 # CONFIG_FB_BROADSHEET is not set
 CONFIG_FB_SSD1289=y
+# CONFIG_EXYNOS_VIDEO is not set
 # CONFIG_BACKLIGHT_LCD_SUPPORT is not set
 # CONFIG_LOGO is not set
 CONFIG_SOUND=y
@@ -1240,7 +1075,7 @@ CONFIG_SND_JACK=y
 # CONFIG_SND_MIXER_OSS is not set
 # CONFIG_SND_PCM_OSS is not set
 # CONFIG_SND_DYNAMIC_MINORS is not set
-# CONFIG_SND_SUPPORT_OLD_API is not set
+CONFIG_SND_SUPPORT_OLD_API=y
 CONFIG_SND_VERBOSE_PROCFS=y
 # CONFIG_SND_VERBOSE_PRINTK is not set
 # CONFIG_SND_DEBUG is not set
@@ -1251,11 +1086,9 @@ CONFIG_SND_VERBOSE_PROCFS=y
 # CONFIG_SND_EMU10K1_SEQ is not set
 # CONFIG_SND_DRIVERS is not set
 # CONFIG_SND_ARM is not set
-# CONFIG_SND_SPI is not set
-# CONFIG_SND_USB is not set
 CONFIG_SND_SOC=y
-CONFIG_SND_LPC313X_SOC=y
-CONFIG_SND_LPC313X_SOC_I2S=y
+CONFIG_SND_LPC31XX_SOC=y
+CONFIG_SND_LPC31XX_SOC_I2S=y
 CONFIG_SND_EA3131_SOC_I2S_UDA1380=y
 CONFIG_SND_USE_EA3131=y
 CONFIG_SND_I2C1_CHANNEL_UDA1380=y
@@ -1277,119 +1110,10 @@ CONFIG_SND_SOC_I2C_AND_SPI=y
 CONFIG_SND_SOC_UDA1380=y
 # CONFIG_SOUND_PRIME is not set
 # CONFIG_HID_SUPPORT is not set
-CONFIG_USB_SUPPORT=y
-CONFIG_USB_COMMON=y
-CONFIG_USB_ARCH_HAS_HCD=y
 # CONFIG_USB_ARCH_HAS_OHCI is not set
 CONFIG_USB_ARCH_HAS_EHCI=y
 # CONFIG_USB_ARCH_HAS_XHCI is not set
-CONFIG_USB=y
-CONFIG_USB_DEBUG=y
-CONFIG_USB_ANNOUNCE_NEW_DEVICES=y
-
-#
-# Miscellaneous USB options
-#
-# CONFIG_USB_DEVICEFS is not set
-CONFIG_USB_DEVICE_CLASS=y
-# CONFIG_USB_DYNAMIC_MINORS is not set
-# CONFIG_USB_OTG_WHITELIST is not set
-# CONFIG_USB_OTG_BLACKLIST_HUB is not set
-CONFIG_USB_MON=y
-# CONFIG_USB_WUSB_CBAF is not set
-
-#
-# USB Host Controller Drivers
-#
-# CONFIG_USB_C67X00_HCD is not set
-CONFIG_USB_EHCI_HCD=y
-# CONFIG_USB_EHCI_EHSET is not set
-CONFIG_USB_EHCI_ROOT_HUB_TT=y
-CONFIG_USB_EHCI_TT_NEWSCHED=y
-# CONFIG_USB_EHCI_MV is not set
-# CONFIG_USB_OXU210HP_HCD is not set
-CONFIG_USB_EHCI_LPC=y
-# CONFIG_USB_ISP116X_HCD is not set
-# CONFIG_USB_ISP1760_HCD is not set
-# CONFIG_USB_ISP1362_HCD is not set
-# CONFIG_USB_SL811_HCD is not set
-# CONFIG_USB_R8A66597_HCD is not set
-
-#
-# USB Device Class drivers
-#
-# CONFIG_USB_ACM is not set
-# CONFIG_USB_PRINTER is not set
-# CONFIG_USB_WDM is not set
-# CONFIG_USB_TMC is not set
-
-#
-# NOTE: USB_STORAGE depends on SCSI but BLK_DEV_SD may
-#
-
-#
-# also be needed; see USB_STORAGE Help for more info
-#
-CONFIG_USB_STORAGE=y
-CONFIG_USB_STORAGE_DEBUG=y
-# CONFIG_USB_STORAGE_REALTEK is not set
-# CONFIG_USB_STORAGE_DATAFAB is not set
-# CONFIG_USB_STORAGE_FREECOM is not set
-# CONFIG_USB_STORAGE_ISD200 is not set
-# CONFIG_USB_STORAGE_USBAT is not set
-# CONFIG_USB_STORAGE_SDDR09 is not set
-# CONFIG_USB_STORAGE_SDDR55 is not set
-# CONFIG_USB_STORAGE_JUMPSHOT is not set
-# CONFIG_USB_STORAGE_ALAUDA is not set
-# CONFIG_USB_STORAGE_ONETOUCH is not set
-# CONFIG_USB_STORAGE_KARMA is not set
-# CONFIG_USB_STORAGE_CYPRESS_ATACB is not set
-# CONFIG_USB_STORAGE_ENE_UB6250 is not set
-# CONFIG_USB_UAS is not set
-# CONFIG_USB_LIBUSUAL is not set
-
-#
-# USB Imaging devices
-#
-# CONFIG_USB_MDC800 is not set
-# CONFIG_USB_MICROTEK is not set
-
-#
-# USB port drivers
-#
-# CONFIG_USB_SERIAL is not set
-
-#
-# USB Miscellaneous drivers
-#
-# CONFIG_USB_EMI62 is not set
-# CONFIG_USB_EMI26 is not set
-# CONFIG_USB_ADUTUX is not set
-# CONFIG_USB_SEVSEG is not set
-# CONFIG_USB_RIO500 is not set
-# CONFIG_USB_LEGOTOWER is not set
-# CONFIG_USB_LCD is not set
-# CONFIG_USB_LED is not set
-# CONFIG_USB_CYPRESS_CY7C63 is not set
-# CONFIG_USB_CYTHERM is not set
-# CONFIG_USB_IDMOUSE is not set
-# CONFIG_USB_FTDI_ELAN is not set
-# CONFIG_USB_APPLEDISPLAY is not set
-# CONFIG_USB_SISUSBVGA is not set
-# CONFIG_USB_LD is not set
-# CONFIG_USB_TRANCEVIBRATOR is not set
-# CONFIG_USB_IOWARRIOR is not set
-# CONFIG_USB_TEST is not set
-# CONFIG_USB_ISIGHTFW is not set
-# CONFIG_USB_YUREX is not set
-# CONFIG_USB_GADGET is not set
-
-#
-# OTG and related infrastructure
-#
-# CONFIG_USB_GPIO_VBUS is not set
-# CONFIG_USB_ULPI is not set
-# CONFIG_NOP_USB_XCEIV is not set
+# CONFIG_USB_SUPPORT is not set
 CONFIG_MMC=y
 # CONFIG_MMC_DEBUG is not set
 # CONFIG_MMC_UNSAFE_RESUME is not set
@@ -1411,36 +1135,9 @@ CONFIG_MMC_LPC31XX=y
 # CONFIG_MMC_SDHCI is not set
 # CONFIG_MMC_SDHCI_PXAV3 is not set
 # CONFIG_MMC_SDHCI_PXAV2 is not set
-# CONFIG_MMC_SPI is not set
 # CONFIG_MMC_DW is not set
-# CONFIG_MMC_VUB300 is not set
-# CONFIG_MMC_USHC is not set
 # CONFIG_MEMSTICK is not set
-CONFIG_NEW_LEDS=y
-CONFIG_LEDS_CLASS=y
-
-#
-# LED drivers
-#
-# CONFIG_LEDS_LM3530 is not set
-CONFIG_LEDS_PCA9532=y
-CONFIG_LEDS_PCA9532_GPIO=y
-# CONFIG_LEDS_GPIO is not set
-# CONFIG_LEDS_LP3944 is not set
-# CONFIG_LEDS_LP5521 is not set
-# CONFIG_LEDS_LP5523 is not set
-# CONFIG_LEDS_PCA955X is not set
-# CONFIG_LEDS_DAC124S085 is not set
-# CONFIG_LEDS_BD2802 is not set
-# CONFIG_LEDS_LT3593 is not set
-# CONFIG_LEDS_RENESAS_TPU is not set
-# CONFIG_LEDS_TCA6507 is not set
-# CONFIG_LEDS_OT200 is not set
-# CONFIG_LEDS_TRIGGERS is not set
-
-#
-# LED Triggers
-#
+# CONFIG_NEW_LEDS is not set
 # CONFIG_ACCESSIBILITY is not set
 CONFIG_RTC_LIB=y
 # CONFIG_RTC_CLASS is not set
@@ -1464,6 +1161,14 @@ CONFIG_CLKDEV_LOOKUP=y
 # Hardware Spinlock drivers
 #
 CONFIG_IOMMU_SUPPORT=y
+
+#
+# Remoteproc drivers (EXPERIMENTAL)
+#
+
+#
+# Rpmsg drivers (EXPERIMENTAL)
+#
 # CONFIG_VIRT_DRIVERS is not set
 # CONFIG_PM_DEVFREQ is not set
 
@@ -1540,17 +1245,6 @@ CONFIG_MISC_FILESYSTEMS=y
 # CONFIG_BEFS_FS is not set
 # CONFIG_BFS_FS is not set
 # CONFIG_EFS_FS is not set
-CONFIG_JFFS2_FS=y
-CONFIG_JFFS2_FS_DEBUG=0
-CONFIG_JFFS2_FS_WRITEBUFFER=y
-# CONFIG_JFFS2_FS_WBUF_VERIFY is not set
-# CONFIG_JFFS2_SUMMARY is not set
-# CONFIG_JFFS2_FS_XATTR is not set
-# CONFIG_JFFS2_COMPRESSION_OPTIONS is not set
-CONFIG_JFFS2_ZLIB=y
-# CONFIG_JFFS2_LZO is not set
-CONFIG_JFFS2_RTIME=y
-# CONFIG_JFFS2_RUBIN is not set
 # CONFIG_LOGFS is not set
 CONFIG_CRAMFS=y
 # CONFIG_SQUASHFS is not set
@@ -1559,10 +1253,9 @@ CONFIG_CRAMFS=y
 # CONFIG_OMFS_FS is not set
 # CONFIG_HPFS_FS is not set
 # CONFIG_QNX4FS_FS is not set
+# CONFIG_QNX6FS_FS is not set
 CONFIG_ROMFS_FS=y
 CONFIG_ROMFS_BACKED_BY_BLOCK=y
-# CONFIG_ROMFS_BACKED_BY_MTD is not set
-# CONFIG_ROMFS_BACKED_BY_BOTH is not set
 CONFIG_ROMFS_ON_BLOCK=y
 # CONFIG_PSTORE is not set
 # CONFIG_SYSV_FS is not set
@@ -1576,6 +1269,7 @@ CONFIG_ROOT_NFS=y
 CONFIG_LOCKD=y
 CONFIG_NFS_COMMON=y
 CONFIG_SUNRPC=y
+# CONFIG_SUNRPC_DEBUG is not set
 # CONFIG_CEPH_FS is not set
 # CONFIG_CIFS is not set
 # CONFIG_NCP_FS is not set
@@ -1674,13 +1368,13 @@ CONFIG_DEBUG_BUGVERBOSE=y
 # CONFIG_DEBUG_CREDENTIALS is not set
 # CONFIG_BOOT_PRINTK_DELAY is not set
 # CONFIG_RCU_TORTURE_TEST is not set
+# CONFIG_RCU_TRACE is not set
 # CONFIG_BACKTRACE_SELF_TEST is not set
 # CONFIG_DEBUG_BLOCK_EXT_DEVT is not set
 # CONFIG_DEBUG_FORCE_WEAK_PER_CPU is not set
 # CONFIG_LKDTM is not set
 # CONFIG_FAULT_INJECTION is not set
 # CONFIG_LATENCYTOP is not set
-# CONFIG_SYSCTL_SYSCALL_CHECK is not set
 # CONFIG_DEBUG_PAGEALLOC is not set
 CONFIG_HAVE_FUNCTION_TRACER=y
 CONFIG_HAVE_FUNCTION_GRAPH_TRACER=y
@@ -1812,16 +1506,21 @@ CONFIG_CRYPTO_HW=y
 #
 CONFIG_BITREVERSE=y
 CONFIG_GENERIC_PCI_IOMAP=y
+CONFIG_GENERIC_IO=y
 CONFIG_CRC_CCITT=y
 # CONFIG_CRC16 is not set
 # CONFIG_CRC_T10DIF is not set
 CONFIG_CRC_ITU_T=y
 CONFIG_CRC32=y
+# CONFIG_CRC32_SELFTEST is not set
+CONFIG_CRC32_SLICEBY8=y
+# CONFIG_CRC32_SLICEBY4 is not set
+# CONFIG_CRC32_SARWATE is not set
+# CONFIG_CRC32_BIT is not set
 # CONFIG_CRC7 is not set
 # CONFIG_LIBCRC32C is not set
 # CONFIG_CRC8 is not set
 CONFIG_ZLIB_INFLATE=y
-CONFIG_ZLIB_DEFLATE=y
 CONFIG_LZO_COMPRESS=y
 CONFIG_LZO_DECOMPRESS=y
 # CONFIG_XZ_DEC is not set
